assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties

```diff
@@ -209,10 +209,14 @@ che.docker.ip.external=NULL
 #   - 'default':      internal address is address of docker host and ephemeral port are used
 #   - 'docker-local': internal address is address of container within docker network, and exposed ports
 #                     are used.
+#   - 'custom': The evaluation strategy may be customized through a template property.
 # The 'docker-local' strategy may be useful if a firewall prevents communication between che-server and
 # workspace containers, but will prevent communication when che-server and workspace containers are not
 # on the same Docker network.
-che.docker.server_evaluation_strategy=default
+che.docker.server_evaluation_strategy=custom
+
+che.docker.custom.external.template=<serverName>.<machineName>.<workspaceId>.<wildcardNipDomain>:<chePort>
+
```

Comment:
I think workspaceName won't be enough, we should include namespace as well (or for you the workspace name includes the namespace)
Because two ppl can have the same workspace name (on different workspace) and in that case it won't work as it will provide host collision